src/unlimited.cpp

```diff
@@ -637,12 +640,12 @@ void UnlimitedSetup(void)
     recvAmt.init("net/recv/total");
     recvAmt.init("net/send/total");
     std::vector<std::string> msgTypes = getAllNetMessageTypes();
-    
+   
     for (std::vector<std::string>::const_iterator i=msgTypes.begin(); i!=msgTypes.end();++i)
-      {
-	new CStatHistory<uint64_t >("net/recv/msg/" +  *i);  // This "leaks" in the sense that it is never freed, but is intended to last the duration of the program.
-	new CStatHistory<uint64_t >("net/send/msg/" +  *i);  // This "leaks" in the sense that it is never freed, but is intended to last the duration of the program.
-      }
+    {
+        mallocedStats.push_front(new CStatHistory<uint64_t >("net/recv/msg/" +  *i));
```

Comment:
the problem is that I don't really know how many message types there will be (although I guess that this should be discoverable at compile time).  if I did know that I could use an array of unique_ptrs as you suggest, but I'm not sure if it makes much difference.